src/App.tsx

```diff
@@ -120,24 +122,20 @@ const App = (): JSX.Element => {
     })();
   }, [bridgeLoaded, loadFaucet]);
 
-  // Maybe loads the vault client - only if the current address is also registered as a vault
-  React.useEffect(() => {
-    if (!bridgeLoaded) return;
-    if (!address) return;
-
-    const id = window.bridge.api.createType(ACCOUNT_ID_TYPE_NAME, address);
-
-    (async () => {
-      try {
-        dispatch(isVaultClientLoaded(false));
-        const vault = await window.bridge.vaults.get(id, RELAY_CHAIN_NATIVE_TOKEN);
-        dispatch(isVaultClientLoaded(!!vault));
-      } catch (error) {
-        // TODO: should add error handling
-        console.log('[App React.useEffect 1] error.message => ', error.message);
-      }
-    })();
-  }, [bridgeLoaded, address, dispatch]);
+  // Detects if connected account is vault operator.
+  const { error: vaultsError } = useQuery<GraphqlReturn<any>, Error>(
+    [GRAPHQL_FETCHER, vaultsByAccountIdQuery(address)],
+    graphqlFetcher<GraphqlReturn<string[]>>(),
+    {
+      enabled: bridgeLoaded && !!address,
+      onSuccess: ({ data: { vaults } }) => {
+        const isVaultOperator = vaults.length > 0;
+        dispatch(isVaultClientLoaded(isVaultOperator));
+      },
+      onError: (error) => console.log('[App useQuery] error.message => ', error.message)
```

Comment:
Could you add the number here? We might not have any other useQuery errors, but would be useful in case we add any in the future.

```suggestion
      onError: (error) => console.log('[App useQuery 1] error.message => ', error.message)
```